streaming/index.js

```diff
@@ -771,40 +583,26 @@ const startServer = async () => {
    * @param {any} req
    * @returns {Promise.<void>}
    */
-  const authorizeListAccess = (listId, req) => new Promise((resolve, reject) => {
+  const authorizeListAccess = async (listId, req) => {
     const { accountId } = req;
 
-    pgPool.connect((err, client, done) => {
-      if (err) {
-        reject();
-        return;
-      }
-
-      // @ts-ignore
-      client.query('SELECT id, account_id FROM lists WHERE id = $1 LIMIT 1', [listId], (err, result) => {
-        done();
-
-        if (err || result.rows.length === 0 || result.rows[0].account_id !== accountId) {
-          reject();
-          return;
-        }
+    const result = await pgPool.query('SELECT id, account_id FROM lists WHERE id = $1 AND account_id = $2 LIMIT 1', [listId, accountId]);
 
-        resolve();
-      });
-    });
-  });
+    if (result.rows.length === 0) {
+      throw new AuthenticationError('List not found');
+    }
+  };
 
   /**
    * @param {string[]} channelIds
    * @param {http.IncomingMessage & ResolvedAccount} req
    * @param {import('pino').Logger} log
    * @param {function(string, string): void} output
-   * @param {undefined | function(string[], SubscriptionListener): void} attachCloseHandler
    * @param {'websocket' | 'eventsource'} destinationType
    * @param {boolean=} needsFiltering
-   * @returns {SubscriptionListener}
+   * @returns {import('./pubsub_manager.js').MessageCallback}
    */
-  const streamFrom = (channelIds, req, log, output, attachCloseHandler, destinationType, needsFiltering = false) => {
+  const streamFrom = (channelIds, req, log, output, destinationType, needsFiltering = false) => {
```

Comment:
This method previously did multiple things:
- created the listener for the messages from Redis
- subscribed to the topics in redis
- attached a close handler in the case of http EventSource connections

This made the code difficult to reason about. It is still currently responsible for sending the message on to the output destination, but that can't really be avoided.

This does mean that for both WebSockets and EventSource connections we need to manage the pubsub manager subscriptions independently, but eventually we'll be able to replace that with a "subscription manager", much like WebSocket's already has, but abstracted for all connection types.

This will mean that processing a message from redis won't need N callbacks, but instead just the one that grabs the subscribed connections and sends the messages, making the streaming server lighter overall (currently we've some GC churn by constantly declaring complex functions such as this.